src/main/java/com/example/b2b_opportunities/Service/PasswordService.java

```diff
@@ -0,0 +1,45 @@
+package com.example.b2b_opportunities.Service;
+
+import com.example.b2b_opportunities.Config.SecurityConfig;
+import com.example.b2b_opportunities.Dto.Request.ResetPasswordDto;
+import com.example.b2b_opportunities.Entity.ConfirmationToken;
+import com.example.b2b_opportunities.Entity.User;
+import com.example.b2b_opportunities.Exception.DisabledUserException;
+import com.example.b2b_opportunities.Exception.PasswordsNotMatchingException;
+import com.example.b2b_opportunities.Repository.ConfirmationTokenRepository;
+import com.example.b2b_opportunities.Repository.UserRepository;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class PasswordService {
+    private final MailService mailService;
+    private final UserRepository userRepository;
+    private final AuthenticationService authenticationService;
+    private final ConfirmationTokenRepository confirmationTokenRepository;
+
+    public String requestPasswordRecovery(String email, HttpServletRequest request) {
+        User user = userRepository.findByEmail(email).orElseThrow(() -> new NullPointerException("User not registered"));
```

Comment:
Why not use User Not found Exception? This throws 500
![image](https://github.com/user-attachments/assets/1c9b70d4-7c9c-455b-9b2b-4427af06dce0)